modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeFetchCPDefinitionByCProductExternalReferenceCodeCheck.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class JavaUpgradeFetchCPDefinitionByCProductExternalReferenceCodeCheck
+	extends BaseJavaTermCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, JavaTerm javaTerm,
+			String fileContent)
+		throws Exception {
+
+		List<String> importNames = javaTerm.getImportNames();
+
+		String content = javaTerm.getContent();
+
+		if (!importNames.contains(
+				"com.liferay.commerce.product.model.CPDefinition")) {
+
+			return content;
+		}
+
+		Matcher methodCallMatcher = _methodCallPattern.matcher(content);
+
+		while (methodCallMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, methodCallMatcher.start());
+
+			if (_checkMethodCall(content, fileContent, methodCall)) {
+				content = StringUtil.replace(
+					content, methodCallMatcher.group(),
+					_reorderParameters(methodCall, methodCallMatcher.group(1)));
+			}
+		}
+
+		return content;
+	}
+
+	@Override
+	protected String[] getCheckableJavaTermNames() {
+		return new String[] {JAVA_METHOD};
+	}
+
+	private boolean _checkMethodCall(
+		String content, String fileContent, String methodCall) {
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
+
+		if (Objects.equals(
+				getVariableTypeName(content, fileContent, parameterList.get(1)),
+				"String") &&
```

Comment:
What happens if a literal string is passed for the `externalReferenceCode`? Would this check work?

Additionally, we might need to check the first argument (`companyId`) as well to make sure it is not some other type/object.

Reply:
I have a question, for example, in the test file the companyId parameter is informed as contextCompany.getCompanyId(), how would I check the type of this variable.

Reply:
For this case, we can't do much and just have to assume that is correct. However, in the case where we are able to determine and get the type (e.g. a variable), then we need to confirm the type.

However, the more I think about it, this particular conditional check might not be necessary. Do you know if there are other methods named `fetchCPDefinitionByCProductExternalReferenceCode()` that have differing parameters?

Reply:
No. This is the only implementation of fetchCPDefinitionByCProductExternalReferenceCode.

I tried to use this variable type check to create a stop condition for the check, otherwise it would go into a loop, since if I put, for example, the number of parameters, it would go into a loop.

Can you think of a better stopping condition?

public CPDefinition fetchCPDefinitionByCProductExternalReferenceCode(String externalReferenceCode, long companyId) throws PortalException;